File: Pro/set-3/30.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    string input;
    cin>>input;
    int num = 0;

    if(input.length()>3)
        num = stoi(input.substr(input.length()-3, input.length()));
    else
        num = stoi(input);
    if(num%8==0)
        cout<<"YES";
    else
        cout<<"NO";
	return 0;
}

File: Pro/set-3/30.py
def checkDivisiblity(x):
  if not int(x)%8:
    return True
  return False

def check(s):
  for i in range(len(s)):
    if checkDivisiblity(s[i]):
      return True
  for i in range(len(s)):
    for j in range(i+1,len(s)):
      if checkDivisiblity(s[i]+s[j]):
        return True
  for i in range(len(s)):
    for j in range(i+1,len(s)):
      for k in range(j+1,len(s)):
        if checkDivisiblity(s[i]+s[j]+s[k]):
          return True
  return False

s = input()
if check(s):
  print("yes")
else:
  print("no")
